app/org/sagebionetworks/bridge/services/ScheduledActivityService.java

```diff
@@ -192,10 +195,14 @@ protected DateTime getDateTime() {
         List<ScheduledActivity> saves = performMerge(scheduledActivities, dbMap);
         activityDao.saveActivities(saves);
         
-        return orderActivities(scheduledActivities, V3_FILTER);
+        List<ScheduledActivity> orderedActivities = orderActivities(scheduledActivities, V3_FILTER);
+        String healthCode = context.getCriteriaContext().getHealthCode();
+        activityEventService.publishActivitiesRetrieved(study, healthCode, DateUtils.getCurrentDateTime());
```

Comment:
We need to publish the event _before_ we schedule them, since the schedule depends on this event. (Same with V4.)

Reply:
Oh boy, very true! Moved.